omero/developers/json-api.txt

```diff
@@ -0,0 +1,526 @@
+a ``limit`` of the first 200 objects by default.
+Pagination can be specified using the ``limit`` and ``offset`` query
+parameters:
+
+::
+
+    # List the first 100 objects (offset=0 by default)
+    <server>/api/v0/m/projects/?limit=100
+
+    # List the next 100 objects
+    <server>/api/v0/m/projects/?limit=100&offset=100
+
+
+Pagination details will be returned in a ``meta`` json object, including
+the ``totalCount`` of objects for that query, the current ``offset`` and
+``limit`` as well as the ``maxLimit`` that you can use.
+
+::
+
+    "meta": {
+        "totalCount": 13240,
+        "maxLimit": 500,
+        "limit": 200,
+        "offset": 0
+    },
+
+Sysadmins can configure the default ``limit`` and ``maxLimit`` settings
+for their server, for example:
+
+::
+
+    $ bin/omero config set omero.web.api.limit 100
+    $ bin/omero config set omero.web.api.max_limit 300
+
+
+Normalizing Experimenters and Groups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When returning a list of JSON objects that each contain ``omero:details`` with
+``owner`` and ``group`` data, these will typically be nested many times
+within the list. In order to avoid this duplication, we can remove objects from
+within each ``omero:details`` and place them under top-level ``experimenters``
+and ``experimenterGroups`` lists.
+You can specify this with the ``?normalize=true`` query parameter.
+NB: Currently this normalizing will only apply to the top-level objects
+being listed, such as Projects, Datasets and Images. Where child objects
+are also loaded (for example Pixels within an Image), the ``omero:details``
+of these objects will not be affected by the ``?normalize=true`` parameter.
+
+
+Child Counts
+^^^^^^^^^^^^
+For container objects such as Projects, Datasets and Screens it is
+often useful to know the number of children within them. This can be
+specified with ``?childCount=true`` parameter.
+This will add an ``omero:childCount`` value to the json data.
+
+
+Filtering by Owner and Group
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Most data in OMERO has an ``Owner`` and is assigned to a permissions
+``Group``. By default, queries will return data from all owners
+accross all groups that are accessible to the current user.
+Use the query strings to filter by owner and/or group:
+
+::
+
+    /api/v0/m/projects/?owner=3&group=5
+
+When you are retrieving data using an object ID you will not need to
+filter by ``group`` since all the data will be in the same group.
+For example, Datasets in a specified Project will all be in the
+same group as the Project.
+
+
+Error handling
+^^^^^^^^^^^^^^
+
+Invalid parameters or invalid JSON will result in a ``400 Bad Request``.
+For example, ``?limit=foo`` will give a response of:
+
+::
+
+    {"message": "invalid literal for int() with base 10: 'foo'"}
+
+
+Unhandled exceptions are handled with a ``500`` error response with an
+error "message" and may also include a "stactrace".
```

Comment:
(I assume here "stac" intentionally doesn't have the "k" I'm used to from elsewhere?)